osu.Game.Modes.Osu/Objects/Slider.cs

```diff
@@ -25,17 +28,66 @@ public override int StackHeight
         }
 
         public double Velocity;
+        public double TickDistance;
 
         public override void SetDefaultsFromBeatmap(Beatmap beatmap)
         {
             base.SetDefaultsFromBeatmap(beatmap);
 
-            Velocity = 100 / beatmap.BeatLengthAt(StartTime, true) * beatmap.BeatmapInfo.BaseDifficulty.SliderMultiplier;
+            var baseDifficulty = beatmap.BeatmapInfo.BaseDifficulty;
+
+            var startBeatLength = beatmap.BeatLengthAt(StartTime);
+            var multipliedStartBeatLength = beatmap.BeatLengthAt(StartTime, true);
+
+            Velocity = 100 / multipliedStartBeatLength * baseDifficulty.SliderMultiplier;
+            TickDistance = (100 * baseDifficulty.SliderMultiplier) / baseDifficulty.SliderTickRate / (multipliedStartBeatLength / startBeatLength);

```

Comment:
Small note: if you are doing double divisions (like here with `startBeatLength`) you can actually also introduce bugs unrelated to readability. Suppose `startBeatLength == 0`.

`a / (1 / startBeatLength) == NaN` whereas `a * startBeatLength = 0`